src/python/pants/bsp/spec/targets.py

```diff
@@ -176,6 +176,7 @@ def to_json_dict(self):
         }
 
 
+@dataclass(frozen=True)
```

Comment:
Eeep. Hope that that wasn't hard to find.

Reply:
Easy enough since without the decorator it has no constructor, so creating an instance failed pretty quickly.